src/main/java/org/apache/sysds/hops/AggBinaryOp.java

```diff
@@ -178,6 +181,11 @@ public Lop constructLops()
 			MMTSJType mmtsj = checkTransposeSelf(); //determine tsmm pattern
 			ChainType chain = checkMapMultChain(); //determine mmchain pattern
 
+			if(mmtsj == MMTSJType.LEFT && input2.isCompressedOutput()){
```

Comment:
you might want to have an additional memory constraint on it, unless the isCompressedOutput selection of input2 would already have done this implicitly; otherwise good